src/app/providers/NFT/BoosterContext.tsx

```diff
@@ -0,0 +1,109 @@
+'use client'
+import { AVERAGE_BLOCKTIME, NFT_BOOSTER_DATA_URL } from '@/lib/constants'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { createContext, ReactNode, useContext, useMemo } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+interface HolderRewards {
+  estimatedRBTCRewards: bigint
+  estimatedRIFRewards: bigint
+  boostedRBTCRewards: bigint
+  boostedRIFRewards: bigint
+}
+
+interface Holders {
+  [holderAddress: string]: HolderRewards
+}
+
+interface BoostData {
+  nftContractAddress: Address
+  boostPercentage: bigint
+  calculationBlock: bigint
+  holders: Holders
+}
+interface NFTBoosterContext {
+  boostData?: BoostData
+  isLoading?: boolean
+  error?: Error | null
+  currentBoost?: HolderRewards
+  hasActiveCampaign?: boolean
+  isBoosted?: boolean
+}
+
+const BoosterContext = createContext<NFTBoosterContext>({
+  boostData: {} as BoostData,
+})
+
+interface BoosterContextProviderProps {
+  children: ReactNode
+}
+export const BoosterProvider = ({ children }: BoosterContextProviderProps) => {
+  const { address } = useAccount()
+  const { data: boostData, isLoading, error, hasActiveCampaign } = useFetchBoostData()
+
+  const value = useMemo(() => {
+    if (!boostData && !isLoading && !error) {
+      return {}
+    }
+
+    if (!address) {
+      return { boostData, isLoading, error, hasActiveCampaign }
+    }
+
+    const currentBoost = boostData?.holders[address]
+    const isBoosted =
+      currentBoost &&
+      (currentBoost.boostedRBTCRewards > 0 ||
+        currentBoost.boostedRIFRewards > 0 ||
+        currentBoost.estimatedRBTCRewards > 0 ||
+        currentBoost.estimatedRIFRewards > 0)
+
+    return { data: boostData, isLoading, error, currentBoost, hasActiveCampaign, isBoosted }
```

Comment:
We dont have any data in the context
```suggestion
    return {boostData, isLoading, error, currentBoost, hasActiveCampaign, isBoosted }
```